lib/spack/spack/spec.py

```diff
@@ -1618,39 +1634,50 @@ def edge_attributes(self) -> str:
 
         union = DependencySpec(parent=Spec(), spec=self, depflag=0, virtuals=())
         all_direct_edges = all(x.direct for x in edges)
+        dep_conditions = set()
 
         for edge in edges:
             union.update_deptypes(edge.depflag)
             union.update_virtuals(edge.virtuals)
+            dep_conditions.add(edge.when)
 
         deptypes_str = ""
         if not all_direct_edges and union.depflag:
             deptypes_str = f"deptypes={','.join(dt.flag_to_tuple(union.depflag))}"
 
         virtuals_str = f"virtuals={','.join(union.virtuals)}" if union.virtuals else ""
-        if not deptypes_str and not virtuals_str:
-            return ""
-        result = f"{deptypes_str} {virtuals_str}".strip()
-        return f"[{result}]"
+
+        conditions = [str(c) for c in dep_conditions if c != Spec()]
+        when_str = f"when='{','.join(conditions)}'" if conditions else ""
+
+        result = " ".join(filter(lambda x: bool(x), (when_str, deptypes_str, virtuals_str)))
+        if result:
+            result = f"[{result}]"
+        return result
 
     def dependencies(
         self,
         name=None,
         deptype: Union[dt.DepTypes, dt.DepFlag] = dt.ALL,
         *,
         virtuals: Optional[Union[str, Sequence[str]]] = None,
+        when: Optional["Spec"] = None,
     ) -> List["Spec"]:
         """Returns a list of direct dependencies (nodes in the DAG)
 
         Args:
             name: filter dependencies by package name
             deptype: allowed dependency types
             virtuals: allowed virtuals
+            when: condition on conditional dependency or Spec() for unconditional
```

Comment:
I find this description really hard to parse.  how about:

```suggestion
            when: optional spec to select matching dependency conditions
```